### PR TITLE
Replace ipcalc with python script

### PR DESCRIPTION
**Pull request by Arvinderpal**
Fixes: https://github.com/metal3-io/metal3-dev-env/issues/224

Output of `./01_prepare_host.sh`:
```
+++ python -c 'import ipaddress; print(ipaddress.ip_network(u"172.22.0.0/24").netmask)'                                                                                                                            
++ export PROVISIONING_NETMASK=255.255.255.0                                                                                                                                                                       
++ PROVISIONING_NETMASK=255.255.255.0 
```

### CODE REVIEW

File: lib/network.sh [hunk 1]
```diff
@@ -18,6 +18,15 @@ function network_address() {
   export resultvar
 }
 
+# REF: https://forum.archive.openwrt.org/viewtopic.php?id=47986&p=1#p220781
+function cdr2mask ()
+{
+   # Number of args to shift, 255..255, first non-255 byte, zeroes
+   set -- $(( 5 - ($1 / 8) )) 255 255 255 255 $(( (255 << (8 - ($1 % 8))) & 255 )) 0 0 0
```
Comment (hardys) [inline lib/network.sh]: This won't work with ipv6 - I think a better approach would be to use python, e.g similar to:

https://github.com/metal3-io/metal3-dev-env/blob/master/lib/network.sh#L41

**Comment by hardys** [comment]
Thanks for the PR - one suggestion which should make this compatible with ipv6

File: lib/network.sh (hunk 1 above)
Comment (Arvinderpal) [reply lib/network.sh]: Ohh. Didn't realize that. Changed it to python. PTAL

**Comment by Arvinderpal** [comment]
I should also note that this does not break anything at the moment since the underlying ubuntu specific code does not make use of `PROVISIONING_NETMASK`. I was walking through the outputs of the metal3-dev-env and happened to see this issue.

**Comment by hardys** [comment]
lgtm subject to CI passing, thanks!

**Comment by maelk** [comment]
This makes the bmo deployment fail

**Comment by hardys** [comment]
> This makes the bmo deployment fail

I don't see many clues in the logs, any idea why?  AFAICS this does generate the same output as the previous ipcalc:

```
$ export PROVISIONING_NETWORK="172.22.0.0/24"
$ ipcalc --netmask "$PROVISIONING_NETWORK" | cut -d= -f2
255.255.255.0
$ export PROVISIONING_NETMASK=${PROVISIONING_NETMASK:-$(python -c "import ipaddress; print(ipaddress.ip_network(u\"$PROVISIONING_NETWORK\").netmask)")}
$ echo $PROVISIONING_NETMASK 
255.255.255.0
```

Not tested this e2e locally though.

**Comment by maelk** [comment]
Due to CentOS changes, this will need a rebase

**Pull request merged**